doc/api/tls.md

```diff
@@ -1094,6 +1094,20 @@ See
 [SSL_get_shared_sigalgs](https://www.openssl.org/docs/man1.1.1/man3/SSL_get_shared_sigalgs.html)
 for more information.
 
+### `tlsSocket.exportKeyingMaterial(length, label)`
+
+* `length` {number}
+  number of bytes to retrieve from keying material
+* `label` {string} an application specific label, typically this will be a
+value from the IANA Exporter Label Registry
+(https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#exporter-labels).
+
+* Returns: {Buffer} {number} of bytes of the keying material
+
+See
+[SSL_export_keying_material](https://www.openssl.org/docs/man1.1.1/man3/SSL_export_keying_material.html)
+for more information.
```

Comment:
added an example

Reply:
The example is good, but it would also be good to include a brief explanation about *why* someone would want to use this API. Doesn't need to be detailed.

Reply:
I'm good with leaving this as is. This is the kind of API that, if you have to ask what you need it for, you don't need it.

Aside: this fits on one line now.

Reply:
Non-blocking, but I'd still prefer a minimal explanation. I generally don't buy into the "if you need to ask you don't need it" explanation.

Reply:
I'm doing hard with finding a minimal explanation, but mostly it's used for security concerns..eap ttls, EAP MS-CHAPv2 etc.. In my case it's used to validate a non encrypted message which is sent as confirmation after an encrypted "tunnel", due to a limitation in the protocol this has been added in this way I guess (see EAP ttls extension: https://tools.ietf.org/html/draft-hanna-eap-ttls-agility-00 ). I could add something like "Keying material is used for validations to prevent differnt kind of attacks in several network protocols, for example in the specifications of IEEE 802.1X"?

Reply:
That would work